src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/AddDebugLogProcessorPass.php

```diff
@@ -22,22 +23,46 @@ public function process(ContainerBuilder $container)
         if (!$container->hasDefinition('profiler')) {
             return;
         }
-        if (!$container->hasDefinition('monolog.logger_prototype')) {
+
+        if ($container->hasDefinition('monolog.logger_prototype') && $container->hasDefinition('debug.log_processor')) {
+            $container->getDefinition('monolog.logger_prototype')
+                ->setConfigurator([__CLASS__, 'configureMonologLogger'])
+                ->addMethodCall('pushProcessor', [new Reference('debug.log_processor')])
+            ;
+
             return;
         }
-        if (!$container->hasDefinition('debug.log_processor')) {
+
+        if (!$container->hasDefinition('logger')) {
             return;
         }
 
-        $definition = $container->getDefinition('monolog.logger_prototype');
-        $definition->setConfigurator([__CLASS__, 'configureLogger']);
-        $definition->addMethodCall('pushProcessor', [new Reference('debug.log_processor')]);
+        $loggerDefinition = $container->getDefinition('logger');
+
+        if (Logger::class === $loggerDefinition->getClass()) {
+            $loggerDefinition
+                ->addMethodCall('enableDebug', [true])
+                ->setConfigurator([__CLASS__, 'configureHttpKernelLogger'])
+            ;
+        }
     }
 
-    public static function configureLogger(mixed $logger)
+    public static function configureMonologLogger(mixed $logger)
     {
-        if (\is_object($logger) && method_exists($logger, 'removeDebugLogger') && \in_array(\PHP_SAPI, ['cli', 'phpdbg'], true)) {
+        if (\is_object($logger) && method_exists($logger, 'removeDebugLogger') && self::shouldDisableDebug()) {
             $logger->removeDebugLogger();
         }
     }
+
+    public static function configureHttpKernelLogger(Logger $logger)
+    {
+        if (self::shouldDisableDebug()) {
+            $logger->enableDebug(false);
+        }
+    }
+
+    private static function shouldDisableDebug(): bool
+    {
+        return \in_array(\PHP_SAPI, ['cli', 'phpdbg'], true);
```

Comment:
let's remove the method and inline this *first* in the checks, so that opcache can optimize the calls

Reply:
Is that what you mean? https://github.com/symfony/symfony/pull/47483/commits/c1a66d6c35debaa6a1a1742abf0e7f7c0ff60cf4